src/platform/platform_linux.cpp

```diff
@@ -38,32 +38,103 @@
 #include "shared/sshfs_server_process_spec.h"
 #include <disabled_update_prompt.h>
 
+#include <QDir>
+#include <QFile>
+#include <QRegularExpression>
+#include <QString>
+#include <QTextStream>
+
+#include <linux/if_arp.h>
+
 namespace mp = multipass;
 namespace mpl = multipass::logging;
 namespace mu = multipass::utils;
 
 namespace
 {
 constexpr auto autostart_filename = "multipass.gui.autostart.desktop";
+constexpr auto category = "Linux platform";
+
+// Fetch the ARP protocol HARDWARE identifier.
+int get_net_type(const QDir& net_dir) // types defined in if_arp.h
+{
+    static constexpr auto default_ret = -1;
+    static const auto type_filename = QStringLiteral("type");
+
+    QFile type_file{net_dir.filePath(type_filename)};
+    if (type_file.open(QIODevice::ReadOnly | QIODevice::Text))
+    {
+        bool ok;
+        auto got = QTextStream{&type_file}.read(6).toInt(&ok); // 6 chars enough for up to 0xFFFF; 0 returned on failure
+        return ok ? got : default_ret;
+    }
+
+    mpl::log(mpl::Level::debug, category, fmt::format("Could not read {}", type_file.fileName()));
```

Comment:
```suggestion
    mpl::log(mpl::Level::warning, category, fmt::format("Could not read {}", type_file.fileName()));
```

Wonder if we should mention that `network-observe` needs to be connected to handle ethernet? Weird, though, that `network-control` doesn't give us that already…

Reply:
Good idea, will do. Yeah, `network-control` did not give us that.

Reply:
Done.